ddl.lua

```diff
@@ -87,24 +75,38 @@ local function check_schema(schema)
     return true
 end
 
-local function set_schema(schema)
+local function check_schema(schema)
     local ok, err = check_schema_format(schema)
     if not ok then
         return nil, err
     end
 
-    local ok, err = check_schema(schema)
-    if not ok then
-        return nil, err
+    if type(box.cfg) == 'function' then
+        return nil, "'box' module isn't configured yet"
+    end
+
+    if box.cfg.read_only then
+        return nil, "Instance is read-only (box.cfg.read_only=true)"
```

Comment:
It's not so. Instance could be in ro mode for several reasons: "orphan" mode or raft state is not a "leader".

Reply:
I suggest to change it to "see box.**info()**.read_only"